### PR TITLE
feat: add update release step

### PR DESCRIPTION
**Pull request by cagiti**
This is a new step designed to update the status (prerelease/release) using the release version, this ties into the version stream work #4120 where we only wish to release a test version of `jx` from the versions stream `jenkins-x-versions`. This step command will allow us to easily toggle the status of a version release.
Enables #4367

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -23,6 +23,7 @@ debug
 #Various temporary created while building/running jx.
 /build/
 release/
+!pkg/cmd/step/update/release/
```
Comment (pmuir) [inline .gitignore]: err, what?

Comment (cagiti) [reply .gitignore]: yeah, that was a quick fix... the step folder is 'release/' what was being ignored by git due to line above

Comment (cagiti) [reply .gitignore]: I'll look to update that to something more sensible

Comment (cagiti) [reply .gitignore]: I meant to go back and change it, but forgot... thanks @pmuir

File: pkg/cmd/step/update/release/step_update_release_github.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+		Aliases: []string{"gh"},
+		Run: func(cmd *cobra.Command, args []string) {
+			options.Cmd = cmd
+			options.Args = args
+			err := options.Run()
+			helper.CheckErr(err)
+		},
+	}
+	AddStepUpdateReleaseFlags(cmd, &options.StepUpdateReleaseOptions)
+	cmd.Flags().BoolVarP(&options.PreRelease, "prerelease", "p", false, "The release state of that version release of the repository")
```
Comment (pmuir) [inline pkg/cmd/step/update/release/step_update_release_github.go]: I don't understand why this is specific to github...

File: pkg/cmd/step/update/release/step_update_release_github.go [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+		return err
+	}
+	gitInfo, err := gits.ParseGitURL(gitURL)
+	if err != nil {
+		return err
+	}
+	o.State.GitInfo = gitInfo
+
+	gitKind, err := o.GitServerKind(gitInfo)
+	gitProvider, err := o.State.GitInfo.CreateProvider(o.InCluster(), authConfigSvc, gitKind, o.Git(), false, o.In, o.Out, o.Err)
```
Comment (pmuir) [inline pkg/cmd/step/update/release/step_update_release_github.go]: This looks wrong. Normally you would do `o.CreateGitProviderFromURLWithoutKind(url)` and that's enough

**Comment by pmuir** [comment]
The use of the git providers is "unusual" and I think the step is not specific to github as it will work with gitea too as of now, but otherwise looks good to me.

**Comment by pmuir** [comment]
The Semantic Pull Request is pending because you have a `wip` PR still in the list

**Pull request merged**